### PR TITLE
Use Mersenne Twister in string_cookie

### PR DESCRIPTION
**Pull request by trshaffer**
This is for #1506 

A CSPRNG might be nice, but I agree it's not really necessary for our use. We wouldn't want to use a `string_cookie` for any sort of key material or anything beyond the simple handshake. It looks like Mersenne Twister was added fairly recently and hasn't made it all the way through the codebase, so thanks for the issue @bbockelm

### CODE REVIEW

File: dttools/src/random.c [hunk 1]
```diff
@@ -18,8 +18,11 @@
 #include <string.h>
 #include <time.h>
 
+int random_initialized = 0;
+
 void random_init (void)
 {
+	if (random_initialized) return;
```
Comment (batrick) [inline dttools/src/random.c]: you can move this int into this funciton as a static variable

Comment (trshaffer) [inline dttools/src/random.c]: Done

File: dttools/src/random.c [hunk 2]
```diff
@@ -33,12 +35,14 @@ void random_init (void)
 	} else {
 		uint64_t seed;
 nasty_fallback:
+		fprintf(stderr, "warning: falling back to low-quality entropy");
```
Comment (dthain) [inline dttools/src/random.c]: Pls use debug(D_NOTICE) here and I will go ahead and merge...

**Comment by trshaffer** [comment]
Fixed, should be ready to merge

**Pull request merged**